scripts/src/modules/wikidata-overlays/wikidata-overlays.js

```diff
@@ -0,0 +1,167 @@
+    Object.entries(data).forEach(([id, item]) => {
+        // Multiple tags can have the same wikidata entry
+        const overlayContainers = document.querySelectorAll(
+            `[data-js-wikidata-overlay="https://www.wikidata.org/wiki/${id}"]`,
+        );
+
+        overlayContainers.forEach((overlayContainer) => {
+            const tagType = overlayContainer.getAttribute(
+                "data-js-wikidata-tag-type-label",
+            );
+
+            const template = createTemplate(item, tagType);
+            overlayContainer.appendChild(template.content);
+        });
+    });
+};
+
+const wikidataLinks = document.querySelectorAll("[data-js-wikidata-url]");
+const wikidataIds = [...wikidataLinks].map((link) =>
+    link.getAttribute("data-js-wikidata-url").split("/").pop(),
+);
+
+/**
+ * Multiple tags can have the same wikidata entry so we remove them before building the API query
+ */
+const uniqueWikidataIds = [...new Set(wikidataIds)];
+
+/**
+ * buildSparqlQuery
+ *
+ * Build query using SPARQL to fetch data from Wikidata
+ * {@link https://www.wikidata.org/wiki/Wikidata:SPARQL_tutorial}
+ *
+ * To assist building the query it's possible to get entity data from the rest API
+ * {@link https://www.wikidata.org/w/rest.php/wikibase/v0/entities/items/[entityId]}
+ *
+ * It's also possible to browse properties by ID
+ * {@link https://hay.toolforge.org/propbrowse/}
+ *
+ * @param {Array} ids
+ * @returns {String}
+ */
+const buildSparqlQuery = (ids) => `
+  # Properties that will be returned by the API
+  SELECT ?item ?label ?description ?image ?countryLabel
+  WHERE {
+    VALUES ?item { ${ids.map((id) => `wd:${id}`).join(" ")} }
+
+    # Specific fields for each item
+    ?item rdfs:label ?label .
+    ?item schema:description ?description .
+    FILTER(LANG(?label) = "en")
+    FILTER(LANG(?description) = "en")
+
+    OPTIONAL { ?item wdt:P18 ?image . }
+
+    OPTIONAL {
+      ?item wdt:P17 ?country .
+      ?country rdfs:label ?countryLabel .
+      FILTER(LANG(?countryLabel) = "en")
+    }
+  }
+`;
+
+const executeSparqlQuery = async (query) => {
+    const url = `https://query.wikidata.org/sparql?query=${encodeURIComponent(query)}&format=json`;
+    const data = await fetchData(url);
+    return data;
+};
+
+const formattedData = {};
+
+const addResultToFormattedData = (result) => {
+    const id = result.item.value.split("/").pop();
+
+    if (!formattedData[id]) {
+        formattedData[id] = {
+            id,
+            label: result.label.value,
+            countryLabel: result?.countryLabel?.value,
+            description: result.description.value,
+            imagePath: result?.image?.value
+                ? `${result.image.value}?width=74`
+                : "",
+        };
+    }
+};
+
+const init = async () => {
+    const sparqlQuery = buildSparqlQuery(uniqueWikidataIds);
+    const data = await executeSparqlQuery(sparqlQuery);
+    const results = data.results.bindings;
```

Comment:
Should we be more defensive about this returned structure? Something like:

```suggestion
    const results = data?.results?.bindings || [];
```